flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/OneSlotPerExecutionSlotAllocator.java

```diff
@@ -0,0 +1,277 @@
+		}
+		return executionVertexSlotRequestIds;
+	}
+
+	private List<SlotExecutionVertexAssignment> createSlotExecutionVertexAssignments(
+			final List<ExecutionVertexSchedulingRequirements> executionVertexSchedulingRequirements,
+			final Map<ExecutionVertexID, SlotRequestId> executionVertexSlotRequestIds) {
+
+		final List<SlotExecutionVertexAssignment> slotExecutionVertexAssignments =
+			new ArrayList<>(executionVertexSchedulingRequirements.size());
+		for (ExecutionVertexSchedulingRequirements schedulingRequirements : executionVertexSchedulingRequirements) {
+			final ExecutionVertexID executionVertexId = schedulingRequirements.getExecutionVertexId();
+			final SlotRequestId slotRequestId = executionVertexSlotRequestIds.get(executionVertexId);
+
+			final SlotExecutionVertexAssignment slotExecutionVertexAssignment =
+				createAndRegisterSlotExecutionVertexAssignment(
+					executionVertexId,
+					new CompletableFuture<>(),
+					throwable -> slotProvider.cancelSlotRequest(slotRequestId, null, throwable));
+			slotExecutionVertexAssignments.add(slotExecutionVertexAssignment);
+		}
+
+		return slotExecutionVertexAssignments;
+	}
+
+	private List<CompletableFuture<PhysicalSlotRequest>> createPhysicalSlotRequestFutures(
+			final List<ExecutionVertexSchedulingRequirements> executionVertexSchedulingRequirements,
+			final Set<ExecutionVertexID> allExecutionVertexIds,
+			final Map<ExecutionVertexID, SlotRequestId> executionVertexSlotRequestIds) {
+
+		final Set<AllocationID> allPreviousAllocationIds =
+			computeAllPriorAllocationIds(executionVertexSchedulingRequirements);
+
+		final List<CompletableFuture<PhysicalSlotRequest>> physicalSlotRequestFutures =
+			new ArrayList<>(executionVertexSchedulingRequirements.size());
+		for (ExecutionVertexSchedulingRequirements schedulingRequirements : executionVertexSchedulingRequirements) {
+			final ExecutionVertexID executionVertexId = schedulingRequirements.getExecutionVertexId();
+			final SlotRequestId slotRequestId = executionVertexSlotRequestIds.get(executionVertexId);
+
+			LOG.debug("Allocate slot with id {} for execution {}", slotRequestId, executionVertexId);
+
+			final CompletableFuture<Collection<TaskManagerLocation>> preferredLocationsFuture =
+				calculatePreferredLocations(
+					executionVertexId,
+					schedulingRequirements.getPreferredLocations(),
+					inputsLocationsRetriever,
+					allExecutionVertexIds);
+
+			// use the task resource profile as the physical slot resource requirement since slot sharing is ignored
+			final CompletableFuture<SlotProfile> slotProfileFuture = preferredLocationsFuture.thenApply(
+				preferredLocations ->
+					SlotProfile.priorAllocation(
+						schedulingRequirements.getTaskResourceProfile(),
+						schedulingRequirements.getTaskResourceProfile(),
+						preferredLocations,
+						Collections.singletonList(schedulingRequirements.getPreviousAllocationId()),
+						allPreviousAllocationIds));
```

Comment:
There is already `ExecutionVertexSchedulingRequirements#physicalSlotResourceProfile` which is the sum of ResourceProfiles of all tasks in the same shared slot.
However, we cannot use it in `OneSlotPerExecutionSlotAllocator` because it ignores slot sharing and the physical slot should fit a single task instead.

Reply:
I meant at least to factor out `getSlotProfileFuture` into a separate private method inside `OneSlotPerExecutionSlotAllocator` and another one inside `DefaultExecutionSlotAllocator `. This would reduce the big method `createPhysicalSlotRequestFutures`.

The next idea was to deduplicate by adding what to use for `physicalSlotResourceProfile` in `SlotProfile.priorAllocation`:
```
AbstractExecutionSlotAllocator#getSlotProfileFuture(..., ResourceProfile physicalSlotResourceProfile)
```
This might be too much.